src/dbcp/data_mart/projects.py

```diff
@@ -535,11 +552,332 @@ def create_long_format(engine: sa.engine.Engine) -> pd.DataFrame:
     )
     _add_derived_columns(long_format)
     pk = ["source", "project_id", "county_id_fips", "resource_clean"]
-    assert long_format.duplicated(subset=pk).sum() == 0, "Duplicate rows in long format"
+    expected_dupes = 2
+    dupes = long_format.duplicated(subset=pk)
+    assert (
+        dupes.sum() == expected_dupes
+    ), f"Expected {expected_dupes} duplicates, found {dupes.sum()}."
     long_format["surrogate_id"] = range(len(long_format))
+
+    # If we only want active projects, grab active projects and remove withdrawn_date and actual_completion_date
+    if active_projects_only:
+        active_long_format = long_format.query("queue_status == 'active'")
+        # drop actual_completion_date and withdrawn_date columns
+        active_long_format = active_long_format.drop(
+            columns=["actual_completion_date", "withdrawn_date"]
+        )
+        return active_long_format
     return long_format
 
 
+def create_geography_change_log(
+    change_log: pd.DataFrame, geography: str = "county_id_fips", freq: str = "Q"
+) -> pd.DataFrame:
+    """Creates a change log of ISO queue projects by geography.
+
+    Each row is a snap shot of the number of projects and capacity in each status and resource class for a given geography and date.
+
+    Currently only includes regions with high coveraage of operational and withdrawn dates: MISO, NYISO, ISONE, PJM, CAISO, SPP.
+    ERCOT will require integrating multiple snapshots of data.
+    """
+    group_keys = [
+        geography,
+        pd.Grouper(key="effective_date", freq=freq),
+        "queue_status",
+        "resource_class",
```

Comment:
I assumed the more general `resource_class` would be more useable because for every resource, we create 6 fields in the change log: number of projects and capacity for each status change. I say we keep it with `resource_class` for now.